app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb

```diff
@@ -20,32 +21,38 @@ def validate_response_exists
           raise Errors::Authentication::AuthnJwt::MissingHttpResponse if @http_response.blank?
         end
 
+        def validate_response_has_a_body
+          raise Errors::Authentication::AuthnJwt::InvalidHttpResponseFormat unless @http_response.respond_to?(:body)
+        end
+
         def create_jwks_from_http_respnse
           @logger.debug(LogMessages::Authentication::AuthnJwt::CreatingJwksFromHttpResponse.new)
 
-          raise Errors::Authentication::AuthnJwt::InvalidHttpResponseFormat unless @http_response.respond_to?(:body)
-
           response_body = @http_response.body
           encoded_body = Base64.encode64(response_body)
           jwks, keys = parse_jwks_response(response_body, encoded_body)
 
-          if keys.blank?
-            raise Errors::Authentication::AuthnJwt::FetchJwksUriKeysNotFound, encoded_body
-          end
-
           @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedJwks.new)
           jwks
         end
 
         def parse_jwks_response(response_body, encoded_body)
-          parsed_response = JSON.parse(response_body)
-          keys = parsed_response['keys']
-          [{ keys: @jwk_set_class.new(keys) }, keys]
-        rescue => e
-          raise Errors::Authentication::AuthnJwt::FailedToConvertResponseToJwks.new(
-            encoded_body,
-            e.inspect
-          )
+          begin
+            parsed_response = JSON.parse(response_body)
+            keys = parsed_response['keys']
+          rescue => e
+            raise Errors::Authentication::AuthnJwt::FailedToConvertResponseToJwks.new(
+              encoded_body,
+              e.inspect
+            )
+          end
+
+          validate_keys_not_empty(keys, encoded_body)
+          return [{ keys: @jwk_set_class.new(keys) }, keys]
```

Comment:
you don't need to return the keys now
```suggestion
          return { keys: @jwk_set_class.new(keys) }, keys]
```

Reply:
Done